tripal_chado/src/Plugin/TripalStorage/ChadoStorage.php

```diff
@@ -589,17 +589,17 @@ protected function buildChadoRecords($values, bool $is_find = FALSE) {
             $table_alias_mapping = array_key_exists('table_alias_mapping', $prop_storage_settings) ? $prop_storage_settings['table_alias_mapping'] : [];
             $path_array = $this->parsePath($field_name, $base_table, $path, $table_alias_mapping, $as);
 
-            // Add to the context.
-            $context['path_string'] = $prop_storage_settings['path'];
-            $context['path_array'] = $path_array;
-
             // We only add joins when the action is 'read_value' because
             // they guarantee a single value (meaning a 1:1 join). For
             // other joins there may be a many to one so we don't want to add
             // those joins off the base table.
             if ($action == 'read_value' and array_key_exists('join', $path_array)) {
               $this->handleJoins($path_array, $context);
```

Comment:
I changed the `handelJoins` function to pass the $path_array be reference. This way the changes to the table aliases get added to the `$context` value that is then used to build the records.